src/core/base.js

```diff
@@ -75,6 +75,10 @@ getJasmineRequireObj().base = function(j$, jasmineGlobal) {
     return obj.nodeType > 0;
   };
 
+  j$.isPromise = function(obj) {
+    return (typeof Promise !== 'undefined') && obj.constructor === Promise;
```

Comment:
I would prefer to not even see `Promise` in here if we can help it. Possibly checking for a `then` function on the object when it doesn't have any keys or something. This would implicitly allow other promise libraries, without the user having to configure anything in Jasmine to make it work.

Does that seem like it would make sense?